### PR TITLE
Date

### PR DESCRIPTION
**Pull request by sarafuentes3025**

### CODE REVIEW

**Comment by aandraca** [comment]
Please next time change the default name of the pull request for something meaningful to the reviewers

File: src/main/java/com/sfuentes/date/DateProgram.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.util.Arrays;
+
+@RequiredArgsConstructor
+public class DateProgram {
+
+  private final String date;
+
+  public String convertDateToWords() {
+
+    var listOfStr = Arrays.asList(date.split("/"));
```
Comment (aandraca) [inline src/main/java/com/sfuentes/date/DateProgram.java:18]: Here you can use the array returned by method split, as far as I can see you are not using any List specific feature.

File: src/test/java/com/sfuentes/date/DateProgramTest.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    DateProgram date = new DateProgram("21/7/2021");
+    assertThat(date.convertDateToWords()).isEqualTo("Wednesday, July 21, 2021");
+  }
+
+  @Test
+  void convertDateToWords_stringDateNotValid_ExceptionThrown() {
+    assertThrows(DateTimeException.class,
+        new DateProgram("21/77/2021")::convertDateToWords, "Date is invalid");
+  }
+}
```
Comment (aandraca) [inline src/test/java/com/sfuentes/date/DateProgramTest.java:23]: Optional.
You can add other two tests to test invalid values in year and day.

**Review by aandraca — CHANGES_REQUESTED** [review]

File: src/main/java/com/sfuentes/date/DateProgram.java (hunk 1 above)
Comment (sarafuentes3025) [reply src/main/java/com/sfuentes/date/DateProgram.java]: Fixed it.

File: src/test/java/com/sfuentes/date/DateProgramTest.java (hunk 2 above)
Comment (sarafuentes3025) [reply src/test/java/com/sfuentes/date/DateProgramTest.java:35]: fixed it.

**Review by aandraca — APPROVED** [review]

**Pull request merged**